packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -79,10 +79,7 @@ const processNodeUrl = (
     s.overwrite(
       node.value!.loc.start.offset,
       node.value!.loc.end.offset,
-      `"${path.posix.join(
-        path.posix.relative(originalUrl, '/'),
-        url.slice(1)
-      )}"`
```

Comment:
I don't understand how this worked, it would end up given a different result depending on cwd? 
@poyoho did you try this PR with the reproduction reported at https://github.com/vitejs/vite/issues/3230 ?

Reply:
when `<script type="module" src="./index.js"></script>` load a module file.
and we had the file tree is 
```
- index.html
- index.js
```

url: http://localhost:3000/env/  import.meta.url is `http://localhost:3000/index.js`

url: http://localhost:3000/  import.meta.url is `http://localhost:3000/index.js`

now I think it is expect. because `import.meta.url` is base on filesystem. `index.html` to `index.js` is `./` so `http://localhost:3000/index.js` I think is expect.

Reply:
In the linked reproduction from the issue, they are setting `base` as `abc` (in the package.json), so both the index.html and index.js should be at `abc/index.html` and `abc/index.js` so `import.meta.url` should be `http://localhost:3000/abc/index.js`, no?

If `base` isn't set, then you are right. But I think this is still an issue 🤔

Reply:
What `./index.js`'s import.meta.url should be when no config.base and request `http://localhost:3000/aaa/index.html`? Vite serve even can request `/a/a/.../.../.../a/a/a/a/` and it default to resolve root's index.html

Reply:
@poyoho maybe it is missing handling the base in that branch

Reply:
Yes, I think when there is a base, you have to think that all your sources are served at that base, so `import.meta.url` should take that into account